extractors/register_parser.py

```
@@ -2,9 +2,154 @@
+
+
+def is_image_based_pdf(file_path):
+    """PDF가 이미지 기반인지 확인"""
+    try:
+        with pdfplumber.open(file_path) as pdf:
+            for page in pdf.pages[:3]:  # 처음 3페이지만 확인
+                text = page.extract_text()
+                if text and len(text.strip()) > 100:  # 충분한 텍스트가 있으면 텍스트 기반
+                    return False
+        return True  # 텍스트가 거의 없으면 이미지 기반
+    except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)